File: go/raindrops/raindrops.go
// Package raindrops implements a Convert function for a variation of FizzBuzz.
package raindrops

import "strconv"

// Convert returns raindrop sounds based on the factors of the input number.
func Convert(n int) (sound string) {
	// Append Pling, Plang, Plong if needed
	if n%3 == 0 {
		sound += "Pling"
	}
	if n%5 == 0 {
		sound += "Plang"
	}
	if n%7 == 0 {
		sound += "Plong"
	}

	// Return digits of n if 3, 5, 7 are not factors, i.e. the string is still empty
	if sound == "" {
		return strconv.Itoa(n)
	}

	return
}

File: python/raindrops/raindrops.py
def convert(number):
    sounds = (
        (3, "Pling"),
        (5, "Plang"),
        (7, "Plong")
    )

    raindrops = "".join(sound for factor, sound in sounds if number % factor == 0)

    return raindrops or str(number)
